collector/collector.go

```diff
@@ -80,15 +80,17 @@ type nodeCollector struct {
 }
 
 // NewNodeCollector creates a new NodeCollector
-func NewNodeCollector() (*nodeCollector, error) {
+func NewNodeCollector(filters map[string]bool) (*nodeCollector, error) {
 	collectors := make(map[string]Collector)
 	for key, enabled := range collectorState {
 		if *enabled {
 			collector, err := factories[key]()
 			if err != nil {
 				return nil, err
 			}
-			collectors[key] = collector
+			if len(filters) == 0 || filters[key] {
```

Comment:
We should error out if someone tries to use a filter which is not loaded. I suggest to not add this code to the collectorState loop but to do another one on top in case any filters are given.